write-fonts/src/tables/glyf.rs

```diff
@@ -939,6 +1170,469 @@ mod tests {
+                }
+            }
+            paths.push(path);
+        }
+        assert_eq!(paths.len(), num_paths);
+        paths
+    }
+
+    fn assert_contour_points(glyph: &SimpleGlyph, all_points: Vec<Vec<CurvePoint>>) {
+        let expected_num_contours = all_points.len();
+        assert_eq!(glyph.contours().len(), expected_num_contours);
+        for (contour, expected_points) in glyph.contours().iter().zip(all_points.iter()) {
+            let points = contour.iter().copied().collect::<Vec<_>>();
+            assert_eq!(points.len(), expected_points.len());
+            assert_eq!(points, *expected_points);
+        }
+    }
+
+    #[test]
+    fn simple_glyphs_from_kurbo_3_lines_closed() {
+        let paths = make_interpolatable_paths(2, "MLLLZ", true);
+        let glyphs = simple_glyphs_from_kurbo(&paths).unwrap();
+
+        assert_contour_points(
+            &glyphs[0],
+            vec![vec![
+                CurvePoint::on_curve(0, 0),
+                CurvePoint::on_curve(1, 1),
+                CurvePoint::on_curve(2, 2),
+            ]],
+        );
+        assert_contour_points(
+            &glyphs[1],
+            vec![vec![
+                CurvePoint::on_curve(3, 3),
+                CurvePoint::on_curve(4, 4),
+                CurvePoint::on_curve(5, 5),
+            ]],
+        );
+    }
+
+    #[test]
+    fn simple_glyphs_from_kurbo_3_lines_implicitly_closed() {
+        let paths = make_interpolatable_paths(2, "MLLZ", false);
+        let glyphs = simple_glyphs_from_kurbo(&paths).unwrap();
+
+        assert_contour_points(
+            &glyphs[0],
+            vec![vec![
+                CurvePoint::on_curve(0, 0),
+                CurvePoint::on_curve(1, 1),
+                CurvePoint::on_curve(2, 2),
+            ]],
+        );
+        assert_contour_points(
+            &glyphs[1],
+            vec![vec![
+                CurvePoint::on_curve(3, 3),
+                CurvePoint::on_curve(4, 4),
+                CurvePoint::on_curve(5, 5),
+            ]],
+        );
+    }
+
+    #[test]
+    fn simple_glyphs_from_kurbo_2_quads_closed() {
+        let paths = make_interpolatable_paths(2, "MQQZ", true);
+        let glyphs = simple_glyphs_from_kurbo(&paths).unwrap();
+
+        assert_contour_points(
+            &glyphs[0],
+            vec![vec![
+                CurvePoint::on_curve(0, 0),
+                CurvePoint::off_curve(1, 1),
+                // CurvePoint::on_curve(2, 2),  // implied oncurve point dropped
+                CurvePoint::off_curve(3, 3),
+            ]],
+        );
+        assert_contour_points(
+            &glyphs[1],
+            vec![vec![
+                CurvePoint::on_curve(4, 4),
+                CurvePoint::off_curve(5, 5),
+                // CurvePoint::on_curve(6, 6),  // implied
+                CurvePoint::off_curve(7, 7),
+            ]],
+        );
+    }
+
+    #[test]
+    fn simple_glyphs_from_kurbo_2_quads_1_line_implictly_closed() {
```

Comment:
general nit: I didn't always find it obvious what each test was testing from the name. This was true across many of the tests. For example, here eliminate_implied_on_curve_between_quads might better describe it?